### DISCUSSION TITLE
Piece of code to generate erDiagram markdown from a collections export

### DISCUSSION THREAD

**Discussion by C0smin** [pocketbase/pocketbase#2252] [Show and tell]
Hi, 

I really appreciate this project and this is my attempt at giving something back. While using it to design some structure to the collections I realized it is becoming a bit complicated to keep a mental map of what is happening. I keep the collections export as code for an attempt at versioning so I built this tiny script to get me a diagram that I can keep in my readme.md for reference.
Made the gist public [here](https://gist.github.com/C0smin/f4ef131b7923e7e191c777821bfbb842) and I'll share the usage here:

```powershell
$hostUri = 'http://localhost:8090'
$cred = Get-Credential -Message "Admin username and password to connect to $hostUri"
$auth = Invoke-RestMethod -Method Post -Uri "$hostUri/api/admins/auth-with-password" -ContentType 'application/json' -Body (@{
        identity = $cred.UserName
        password = $cred.GetNetworkCredential().Password
    } | ConvertTo-Json)
$jsonObject = (Invoke-RestMethod -Method Get -Uri "$hostUri/api/collections?page=1&perPage=100&sort=updated" -Headers @{ Authorization = $auth.token }).items
$markdown = Export-PocketBaseErDiagramMarkdown -CollectionsExport $jsonObject
# $markdown | Out-File -FilePath diagram.md -NoNewline # Export to current path as diagram.md
# $markdown | Set-Clipboard # Export to clipboard to put wherever you want
```

I visualize and tested this using VSCode with [mermaid support](https://marketplace.visualstudio.com/items?itemName=bierner.markdown-mermaid).

**Comment by icflorescu** [comment]
...or you could use [pocketbase-uml.github.io](https://pocketbase-uml.github.io/) 😊

[![PocketBaseUML](https://user-images.githubusercontent.com/581999/232122782-c1210ee1-b761-42f1-b806-cc9335167ff4.png)](https://pocketbase-uml.github.io/)

I tried using mermaid for the project initially, but wasn't exactly happy with the generated diagrams, so in the end I went with a custom version of nomnoml.
